### ISSUE TITLE
edit conflict is confusing for mappers

### ISSUE THREAD

**Issue by matkoniecz** [bryceco/GoMap#686]
User gets confusing message and no clear way to handle it.

I would propose to 

- add note "It means that objects you edited were edited also by someone else"
- Add button "throw away edits and cached data" to provide an escape hatch for someone who is not an expert.
- Also, throw away cached data for this area (or all) after edit conflict appears and then upload is done (because user resolved it manually or via undoing edits until they could be uploaded)

I was mapping with someone who works in IT and edit conflict message confused them, they were able to resolve it only because I realized where we conflicted. And then they run into edit conflict again because GoMap has not downloaded new data after they undo some edits and uploaded nonconflicting ones.

PS: GoMap works great for them, I finally convinced them to start mapping :)

**Comment by matkoniecz** [comment]
Yeah, resolving edit conflicts would be great but I am not even going to propose adding it without being willing to implement that.

**Comment by bryceco** [comment]
There's a rudimentary conflict resolution in place already: if you try to upload something and the server says there is a newer version then we discard your version, download the server's version, and then retry the upload. It will do that multiple times, but there are still cases where that isn't sufficient, so that's where it could be improved.

**Comment by matkoniecz** [comment]
> then we discard your version, download the server's version, and then retry the upload

is it happening automatically? Or should user reupload after getting this error?

**Comment by bryceco** [comment]
That behavior occurs silently and automatically. It might be buggy or not work correctly in some cases. 

If you want to try testing some scenarios on the dev server you can enter "dev" as the server URL in Advanced Settings.

**Comment by matkoniecz** [comment]
It failed in case where node was attached to modified way. Maybe this process got confused because it needed to detach node and give up on gluing it to a road?

> That behavior occurs silently and automatically.

Maybe warning user that some part of edit were thrown away would be a good idea

(phrased as "There was edit conflict, as someone else also edited - but editor automatically rescued part of the edit!")

**Comment by Udarthegreat** [comment]
> It failed in case where node was attached to modified way. Maybe this process got confused because it needed to detach node and give up on gluing it to a road?

I also ran into similar issue about a week back when I was trying to commit [144915276](https://www.openstreetmap.org/changeset/144915276) because I had attached the following [way](https://www.openstreetmap.org/way/1229747103) to the hotel and had in commit [144915577](https://www.openstreetmap.org/changeset/144915577) updated that same feature with a different editor and had committed that commit right before I tried to upload [144915276](https://www.openstreetmap.org/changeset/144915276) without thinking of the possible change set conflict. I eventually figured out what was causing the error, went back through my undo history untill it was fixed and then added everything I had done back other than connecting those two. This got me thinking on a possible way to fix some of these kinds of issues, what if when you get an error from the OSM api that amounts of change set conflicts and cannot fix it through the silent background process you already do, you could add a small button with an error icon of some sort above the change set comment input field that when pressed asks the user if they want to completely revert the way to what ever its current status is on the OSM servers. Like this the user would not have to go back and then redo possibly allot of work, you would just lose what ever you did to that one way/node/relation. All it would need to do it figure out what changes the user has done to the item in question and purge it from memory and then request the current state of the way on the OSM servers and set it as that locally since we know our knowledge of the way is out of date at this point. 

These are just my ideas on the matter and since this was anyways open I figured it couldn't hurt to put my ideas on this out there, obviously this is a complex issue and I most definitely don't have the knowledge or resources (a requisite device) to put anything together for IOS devices, but I hope that this does help with however this issue will eventually be fixed. I have also attached below 2 images were I have drawn on top of the commit pane what I mean with two possible positions were the blue error is option 1 and the green arrow option 2. (If the image is to big I can make it smaller). For the second the number (1 in this case) would likely change with the number of conflicting features. These are just simple examples I put together quickly.

<img src="https://github.com/bryceco/GoMap/assets/29469837/bdbcb0e2-aa04-4bb6-a691-323fafb42b18" width="350" />

<img src="https://github.com/bryceco/GoMap/assets/29469837/bdbcb0e2-aa04-4bb6-a691-323fafb42b18" width="350" />
